### DISCUSSION TITLE
How to launch bash and source a script from kitty’s startup session

### DISCUSSION THREAD

**Discussion by jenstroeger** [kovidgoyal/kitty#6309] [Q&A]
Hello,

In an exitsing bash, running
```sh
/opt/local/bin/bash --init-file ~/path/to/project/.venv/bin/activate
```
works and Python’s virtual environment is activated.

However, running that same line from the startup session script:
```sh
launch --cwd=~/path/to/project /opt/local/bin/bash --init-file ~/path/to/project/.venv/bin/activate
# Doesn’t work either: launch --cwd ~/path/to/project /opt/local/bin/bash --init-file .venv/bin/activate
```
runs the shell but does not activate the virtual environment. What am I missing?

Much thanks!

**Comment by kovidgoyal** [comment]
Use absolute paths, dont use ~

**Reply by jenstroeger** [reply]
Nope, doesn’t work:
```
launch --cwd=/abs/path/to/project /opt/local/bin/bash --init-file /abs/path/to/project/.venv/bin/activate
```

**Reply by kovidgoyal** [reply]
Remember to pass -i and -l to bash to make it source your rc files.

**Reply by jenstroeger** [reply]
@kovidgoyal no dice 😞

I’ll look into this tomorrow, unless you have a working example at hand…

**Reply by kovidgoyal** [reply]
No clue I dont use venvs. I can tell you that 
```sh
kitty -o 'map f1 launch --hold bash --init-file /tmp/test -i'
```
with echo hellow world in /tmp/test
works fine.
